WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/details/editing/address/AddressViewModelTest.kt

```diff
@@ -0,0 +1,205 @@
+package com.woocommerce.android.ui.orders.details.editing.address
+
+import androidx.lifecycle.Observer
+import androidx.lifecycle.SavedStateHandle
+import com.woocommerce.android.model.Location
+import com.woocommerce.android.tools.SelectedSite
+import com.woocommerce.android.ui.orders.details.editing.address.AddressViewModel.ViewState
+import com.woocommerce.android.viewmodel.BaseUnitTest
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runBlockingTest
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.wordpress.android.fluxc.model.data.WCLocationModel
+import org.wordpress.android.fluxc.store.WCDataStore
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+@ExperimentalCoroutinesApi
+class AddressViewModelTest : BaseUnitTest() {
+
+    private val savedStateHandle: SavedStateHandle = SavedStateHandle()
+
+    private val selectedSite: SelectedSite = mock()
+
+    private val location = WCLocationModel().apply {
+        name = "Brazil"
+        code = "BR"
+    }
+
+    private val dataStore: WCDataStore = mock()
+
+    private val addressViewModel = AddressViewModel(savedStateHandle, selectedSite, dataStore)
+
+    private val viewStateObserver: Observer<ViewState> = mock()
+
+    @Before
+    fun setup() {
+        addressViewModel.viewStateData.liveData.observeForever(viewStateObserver)
+    }
+
+    @Test
+    fun `Should fetch countries and states on start if they've never been fetched`() {
+        whenever(dataStore.getCountries()).thenReturn(emptyList())
+        coroutinesTestRule.testDispatcher.runBlockingTest {
+            addressViewModel.start(
+                countryCode = "countryCode",
+                stateCode = "stateCode"
+            )
+            verify(dataStore, times(1)).fetchCountriesAndStates(selectedSite.get())
```

Comment:
⛏️  minor nitpick: `verify` by default checks for exactly 1 execution
```suggestion
            verify(dataStore).fetchCountriesAndStates(selectedSite.get())
```